### PR TITLE
Reduce queries to VACOLS::Staff

### LINKED ISSUE #6040: Ensure `LegacyTasks#index` doesn't trigger an unnecessary amount of staff table queries
While investigating `/queue` load times for a user with 100+ appeals assigned to them, a flurry of VACOLS Staff table queries showed up in the logs. A test showed that these staff table queries accounted for 7 seconds of the total 11 second response time for this user. It's possible that we could make pageloads much faster for users with significant # of appeals assigned to them by optimizing.

It may be the case that we're accidentally making a different query for each appeal assigned to a user, while we could batch these queries instead.

The flurry of queries appear to be happening somewhere in the request after this line— maybe in the serializers?
https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/repositories/queue_repository.rb#L32

```
2018-06-27 17:04:41 -0400]   VACOLS::Staff Load (78.1ms)  SELECT  "VACOLS"."STAFF".* FROM "VACOLS"."STAFF" WHERE "VACOLS"."STAFF"."SLOGID" = :a1 AND ROWNUM <= :a2   [["slogid", nil], ["LIMIT", 1]]
[2018-06-27 17:04:41 -0400]   LegacyAppeal Load (1.1ms)  SELECT  "legacy_appeals".* FROM "legacy_appeals" WHERE "legacy_appeals"."vacols_id" = $1 LIMIT $2  [["vacols_id", "3358065"], ["LIMIT", 1]]
[2018-06-27 17:04:41 -0400]   VACOLS::Staff Load (77.4ms)  SELECT  "VACOLS"."STAFF".* FROM "VACOLS"."STAFF" WHERE "VACOLS"."STAFF"."SLOGID" = :a1 AND ROWNUM <= :a2   [["slogid", nil], ["LIMIT", 1]]
[2018-06-27 17:04:41 -0400]   LegacyAppeal Load (1.1ms)  SELECT  "legacy_appeals".* FROM "legacy_appeals" WHERE "legacy_appeals"."vacols_id" = $1 LIMIT $2  [["vacols_id", "3362414"], ["LIMIT", 1]]
[2018-06-27 17:04:41 -0400]   VACOLS::Staff Load (77.3ms)  SELECT  "VACOLS"."STAFF".* FROM "VACOLS"."STAFF" WHERE "VACOLS"."STAFF"."SLOGID" = :a1 AND ROWNUM <= :a2   [["slogid", nil], ["LIMIT", 1]]
[2018-06-27 17:04:41 -0400]   LegacyAppeal Load (1.2ms)  SELECT  "legacy_appeals".* FROM "legacy_appeals" WHERE "legacy_appeals"."vacols_id" = $1 LIMIT $2  [["vacols_id", "3306129"], ["LIMIT", 1]]
[2018-06-27 17:04:41 -0400]   VACOLS::Staff Load (77.5ms)  SELECT  "VACOLS"."STAFF".* FROM "VACOLS"."STAFF" WHERE "VACOLS"."STAFF"."SLOGID" = :a1 AND ROWNUM <= :a2   [["slogid", nil], ["LIMIT", 1]]
[2018-06-27 17:04:41 -0400]   LegacyAppeal Load (1.1ms)  SELECT  "legacy_appeals".* FROM "legacy_appeals" WHERE "legacy_appeals"."vacols_id" = $1 LIMIT $2  [["vacols_id", "3229940"], ["LIMIT", 1]]
[2018-06-27 17:04:41 -0400]   VACOLS::Staff Load (77.2ms)  SELECT  "VACOLS"."STAFF".* FROM "VACOLS"."STAFF" WHERE "VACOLS"."STAFF"."SLOGID" = :a1 AND ROWNUM <= :a2   [["slogid", nil], ["LIMIT", 1]]
[2018-06-27 17:04:41 -0400]   LegacyAppeal Load (1.2ms)  SELECT  "legacy_appeals".* FROM "legacy_appeals" WHERE "legacy_appeals"."vacols_id" = $1 LIMIT $2  [["vacols_id", "3209922"], ["LIMIT", 1]]
[2018-06-27 17:04:42 -0400]   VACOLS::Staff Load (78.1ms)  SELECT  "VACOLS"."STAFF".* FROM "VACOLS"."STAFF" WHERE "VACOLS"."STAFF"."SLOGID" = :a1 AND ROWNUM <= :a2   [["slogid", nil], ["LIMIT", 1]]
[2018-06-27 17:04:42 -0400]   LegacyAppeal Load (1.1ms)  SELECT  "legacy_appeals".* FROM "legacy_appeals" WHERE "legacy_appeals"."vacols_id" = $1 LIMIT $2  [["vacols_id", "3205000"], ["LIMIT", 1]]
[2018-06-27 17:04:42 -0400]   VACOLS::Staff Load (77.5ms)  SELECT  "VACOLS"."STAFF".* FROM "VACOLS"."STAFF" WHERE "VACOLS"."STAFF"."SLOGID" = :a1 AND ROWNUM <= :a2   [["slogid", nil], ["LIMIT", 1]]
```

# AC
- Investigate this situation and run one staff table query instead of N queries if possible.

### PR DESCRIPTION
**Pull request by dannysellers**
Resolves #6040 

Using [active-record-query-trace](https://github.com/ruckus/active-record-query-trace), I got:
```
[2018-07-03 13:46:10 -0400]   VACOLS::Staff Load (4.6ms)  SELECT  "VACOLS"."STAFF".*
 FROM "VACOLS"."STAFF" WHERE "VACOLS"."STAFF"."SLOGID" = :a1 FETCH FIRST :a2 ROWS ONLY
[["slogid", nil], ["LIMIT", 1]] [localhost] [6de22403-6db0-45ce-94b9-1913f4bb0d5a]
[localhost] [BVAAABSHIRE] Query Trace > app/services/appeal_repository.rb:113:in `build_appeal'
  from app/repositories/queue_repository.rb:35:in `block (2 levels) in appeals_from_tasks'
  from app/repositories/queue_repository.rb:34:in `map'
  from app/repositories/queue_repository.rb:34:in `block in appeals_from_tasks'
  from app/services/metrics_service.rb:13:in `block in record'
```

Within `AppealRepository.build_appeal`, we call `set_vacols_values`, which accesses the Folder's `outcoder`. [`VACOLS::Folder.outcoder`](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/vacols/folder.rb#L6) is a foreign key to Staff (on `:tiocuser`), causing a VACOLS query with `slogid: nil`.

Without the patch, loading `/queue` as `BVAAABSHIRE` results in 24 Staff queries. With the patch, 5.

In UAT VACOLS, there are 1,795,312 Folders with a null outcoder, and 474,134 without; in local FACOLS, there're 62 null `tiocuser` Folders, 21 non-null.

It's unclear what causes this behavior (querying a table when the foreign key is `nil`), or if there's some way we've defined [the relation](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/vacols/folder.rb#L6) that could be a factor. The closest issue I've found is https://github.com/rails/rails/issues/18312, which concerns running a query on a `has_one` foreign key if the key is `nil`, but only when using [`references`](https://apidock.com/rails/ActiveRecord/QueryMethods/references).

### Acceptance Criteria 
- [ ] Tests pass, site functions
- [ ] Fewer queries when loading/constructing `LegacyAppeal`s

### Testing Plan
1. Go to `/queue` with user who has a lot of cases, confirm fewer Staff queries

### CODE REVIEW

File: app/services/appeal_repository.rb [hunk 1]
```diff
@@ -119,7 +119,7 @@ def self.build_appeal(case_record, persist = false)
   def self.set_vacols_values(appeal:, case_record:)
     correspondent_record = case_record.correspondent
     folder_record = case_record.folder
-    outcoder_record = folder_record.outcoder
+    outcoder_record = folder_record.outcoder if folder_record.tiocuser?
```
Comment (amprokop) [inline app/services/appeal_repository.rb]: Worth adding a short comment? `Only fetch the outcoder from the staff table if the key to that table is not nil`

**Comment by amprokop** [comment]
👏

File: app/services/appeal_repository.rb (hunk 1 above)
Comment (evankroske) [reply app/services/appeal_repository.rb]: Maybe I missed something, but I don't see the comment in this PR. Is it being added in a future PR?

Comment (dannysellers) [reply app/services/appeal_repository.rb]: Yup, #6109

**Pull request merged**